### PR TITLE
KT-42652 parcelize fix codegen for generic classes

### PR DESCRIPTION
**Pull request by zveznicht**
When Parcelable class has generics parcelize plugin generates Creator methods for them with type arguments, which are not available in Creator instance.  
To fix this I replace them with wildcard. 

While trying to write test to reproduce this issues I got error in IR variant of parcelize codegen. So I have to add change there too.

### CODE REVIEW

File: compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/builders/declarations/declarationBuilders.kt [hunk 1]
```diff
@@ -164,6 +165,7 @@ fun IrClass.addFunction(
         this.origin = origin
     }.apply {
         if (!isStatic) {
+            copyTypeParametersFrom(parentAsClass)
```
Comment (zveznicht) [inline compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/builders/declarations/declarationBuilders.kt:168]: without this I got exception later in _remapTypeParameters_

Comment (udalov) [reply compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/builders/declarations/declarationBuilders.kt:168]: This change doesn't look correct. Could you clarify what exception were you getting?

Comment (zveznicht) [reply compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/builders/declarations/declarationBuilders.kt:168]:
```
java.lang.IndexOutOfBoundsException: Empty list doesn't contain element at index 0.
	at kotlin.collections.EmptyList.get(Collections.kt:36)
	at kotlin.collections.EmptyList.get(Collections.kt:24)
	at org.jetbrains.kotlin.backend.common.ir.IrUtilsKt.remapTypeParameters(IrUtils.kt:328)
	at org.jetbrains.kotlin.backend.common.ir.IrUtilsKt.remapTypeParameters(IrUtils.kt:341)
	at org.jetbrains.kotlin.backend.common.ir.IrUtilsKt.copyTo$default(IrUtils.kt:119)
	at org.jetbrains.kotlin.ir.builders.declarations.DeclarationBuildersKt.addFunction(declarationBuilders.kt:169)
	at org.jetbrains.kotlin.ir.builders.declarations.DeclarationBuildersKt.addFunction$default(declarationBuilders.kt:144)
	at org.jetbrains.kotlin.parcelize.ir.ParcelizeIrTransformer.addOverride(ParcelizeIrTransformer.kt:279)
	at org.jetbrains.kotlin.parcelize.ir.ParcelizeIrTransformer.visitClass(ParcelizeIrTransformer.kt:115)
	at org.jetbrains.kotlin.ir.visitors.IrElementVisitorVoid$DefaultImpls.visitClass(IrElementVisitorVoid.kt:44)
	at org.jetbrains.kotlin.parcelize.ir.ParcelizeIrTransformer.visitClass(ParcelizeIrTransformer.kt:42)
	at org.jetbrains.kotlin.parcelize.ir.ParcelizeIrTransformer.visitClass(ParcelizeIrTransformer.kt:42)
```

Comment (zveznicht) [reply compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/builders/declarations/declarationBuilders.kt:168]: btw what is the right place for more 'unit' tests for IR builders?

Comment (zveznicht) [reply compiler/ir/backend.common/src/org/jetbrains/kotlin/ir/builders/declarations/declarationBuilders.kt]: replaced with explicit passing type argument to copyTo

**Pull request merged**